r/src/embedding.r

```diff
@@ -20,32 +20,55 @@ runEmbedding <- function(req) {
     type <- req$body$type
     config <- req$body$config
     pca_nPCs <- 30 
+
     if (type == "pca") {
         # Leaving this here to add parameters in the future. Won't leave uncommented to avoid recalculating PCA>
         # RunPCA(data, npcs = 50, features = VariableFeatures(object=data), verbose=FALSE)
         df_embedding <- Embeddings(data, reduction = type)[,1:2]
     } else if(type=="tsne"){
-        data <- RunTSNE(data,
+        # Embedding was computed in data processing. Just to avoid possible change in dimensional reduction technique, 
+        # I leave this code to check if it has not been computed. 
+        if(!"tsne"%in%names(data@reductions))
+            data <- RunTSNE(data,
```

Comment:
I don't see where `data` is first defined should it be comming from function parameter?

Reply:
It is a global varaible. Once the platform is running we need to create a ticket for re-organizing the worker tasks